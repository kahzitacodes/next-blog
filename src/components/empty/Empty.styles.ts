import tw from 'tailwind-styled-components'

export const Wrapper = tw.div`
  flex
  flex-col
  items-center
  justify-center
  gap-6
  min-h-[calc(100vh_-_30rem)]
`

export const Content = tw.div`
  flex
  flex-col
  items-center
  justify-center
  gap-6
`
