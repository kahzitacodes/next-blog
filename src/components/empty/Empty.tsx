import { Button } from '../button'
import { Icon } from '../icon'
import { PropTypes } from './Empty.types'
import * as S from './Empty.styles'

export const Empty = (props: PropTypes) => {
  const { description, title, button } = props

  return (
    <S.Wrapper {...props}>
      <Icon name="Rat" size={80} strokeWidth={1} />
      <S.Content className="flex flex-col items-center justify-center gap-2">
        <h1 className="text-3xl font-bold">{title}</h1>
        <p className="text-zinc-400">{description}</p>
      </S.Content>
      {button ? <Button {...button} /> : null}
    </S.Wrapper>
  )
}
