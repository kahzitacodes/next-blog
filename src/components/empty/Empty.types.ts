import { ButtonTypes } from '../button'

export type PropTypes = {
  title: string
  description: string
  button?: ButtonTypes
}
