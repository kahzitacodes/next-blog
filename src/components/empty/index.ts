export * from './Empty'
export type { PropTypes as EmptyProps } from './Empty.types'
