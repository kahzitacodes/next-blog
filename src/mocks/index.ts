export * from './navigation'
export * from './profile'
export * from './social-links'
export * from './post'
export * from './pre'
export * from './pagination'
