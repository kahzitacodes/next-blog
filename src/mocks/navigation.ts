import { NavItem } from '@/types'

export const navigationMock: NavItem[] = [
  {
    title: 'Home',
    href: '/'
  },
  {
    title: 'About',
    href: '/about'
  },
  {
    title: 'Contact',
    href: '/contact'
  }
]
