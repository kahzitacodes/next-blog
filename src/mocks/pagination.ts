export const paginationMock = {
  currentPage: 1,
  totalPages: 10
}
