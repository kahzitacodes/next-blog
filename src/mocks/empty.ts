import { EmptyProps } from '@/components/empty'

export const emptyMock: EmptyProps = {
  title: 'Oops!',
  description: 'Page not found'
}
