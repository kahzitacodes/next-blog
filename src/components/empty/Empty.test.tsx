import { fireEvent, render, screen } from '@testing-library/react'
import { Empty } from '.'
import { emptyMock } from '@/mocks/empty'

describe('<Empty />', () => {
  it('should render successfully', () => {
    const { container } = render(<Empty {...emptyMock} />)

    expect(screen.getByRole('heading', { name: /Oops!/i })).toBeInTheDocument()
    expect(screen.getByText(/page not found/i)).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render button anc call its function', () => {
    const handleBack = jest.fn()
    const { container } = render(
      <Empty
        {...emptyMock}
        button={{ children: 'Back to list', onClick: handleBack }}
      />
    )

    const button = screen.getByRole('button', { name: /back to list/i })

    expect(button).toBeInTheDocument()

    fireEvent.click(button)
    expect(handleBack).toHaveBeenCalled()

    expect(container.firstChild).toMatchSnapshot()
  })
})
