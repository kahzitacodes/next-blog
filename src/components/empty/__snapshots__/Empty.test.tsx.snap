// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Empty /> should render button anc call its function 1`] = `
<div
  button="[object Object]"
  class="flex flex-col items-center justify-center gap-6 min-h-[calc(100vh_-_30rem)]"
  description="Page not found"
  title="Oops!"
>
  <svg
    class="lucide lucide-rat"
    data-testid="Rat"
    fill="none"
    height="80"
    stroke="currentColor"
    stroke-linecap="round"
    stroke-linejoin="round"
    stroke-width="1"
    viewBox="0 0 24 24"
    width="80"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M13 22H4a2 2 0 0 1 0-4h12"
    />
    <path
      d="M13.236 18a3 3 0 0 0-2.2-5"
    />
    <path
      d="M16 9h.01"
    />
    <path
      d="M16.82 3.94a3 3 0 1 1 3.237 4.868l1.815 2.587a1.5 1.5 0 0 1-1.5 2.1l-2.872-.453a3 3 0 0 0-3.5 3"
    />
    <path
      d="M17 4.988a3 3 0 1 0-5.2 2.052A7 7 0 0 0 4 14.015 4 4 0 0 0 8 18"
    />
  </svg>
  <div
    class="flex flex-col items-center justify-center gap-2"
  >
    <h1
      class="text-3xl font-bold"
    >
      Oops!
    </h1>
    <p
      class="text-zinc-400"
    >
      Page not found
    </p>
  </div>
  <button
    class="flex items-center gap-1 text-white font-medium h-[40px] px-4 rounded-md w-fit transition-colors duration-200 group disabled:text-zinc-600 disabled:cursor-default disabled:border-zinc-600 bg-transparent border-2 border-white hover:border-link hover:text-link disabled:pointer-events-none"
    variant="secondary"
  >
    Back to list
  </button>
</div>
`;

exports[`<Empty /> should render successfully 1`] = `
<div
  class="flex flex-col items-center justify-center gap-6 min-h-[calc(100vh_-_30rem)]"
  description="Page not found"
  title="Oops!"
>
  <svg
    class="lucide lucide-rat"
    data-testid="Rat"
    fill="none"
    height="80"
    stroke="currentColor"
    stroke-linecap="round"
    stroke-linejoin="round"
    stroke-width="1"
    viewBox="0 0 24 24"
    width="80"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M13 22H4a2 2 0 0 1 0-4h12"
    />
    <path
      d="M13.236 18a3 3 0 0 0-2.2-5"
    />
    <path
      d="M16 9h.01"
    />
    <path
      d="M16.82 3.94a3 3 0 1 1 3.237 4.868l1.815 2.587a1.5 1.5 0 0 1-1.5 2.1l-2.872-.453a3 3 0 0 0-3.5 3"
    />
    <path
      d="M17 4.988a3 3 0 1 0-5.2 2.052A7 7 0 0 0 4 14.015 4 4 0 0 0 8 18"
    />
  </svg>
  <div
    class="flex flex-col items-center justify-center gap-2"
  >
    <h1
      class="text-3xl font-bold"
    >
      Oops!
    </h1>
    <p
      class="text-zinc-400"
    >
      Page not found
    </p>
  </div>
</div>
`;
